### DISCUSSION TITLE
GIT password not getting encripted in <project directory>/.git/conf

### DISCUSSION THREAD

**Discussion by balakrishna-maduru** [eclipse-theia/theia#12230] [Q&A]
Hi Team,

When we issue git clone https://username:password@hostname/repo.git the user name and password is storing directly in .git/conf file, is there any way we can make it encripted and decode password while doing git operations like push merge ect..

Please help me if any such functionality  available in Theia already?

**Comment by msujew** [comment]
Hey @balakrishna-maduru,

In the end, Theia is just using the `git` executable under the hood to provide git functionality. We rely on [dugite](https://github.com/desktop/dugite) to provide high-level git bindings. As far as I know, git does never encrypt the data in your config in any way.

In theory you can hook into the integration on our side (see [this file](https://github.com/eclipse-theia/theia/blob/master/packages/git/src/node/dugite-git.ts)) and decrypt and re-encrypt the file every time it is used. However, crafty users will probably still be able to figure out the password while it is decrypted. Some git commands might take a while to complete, during which time the password is still visible in plain text.

It's probably safest to use other options instead, such as disposable SSH-keys or something similar. I know that [gitpod](https://github.com/gitpod-io/gitpod) is using a secure way to interact with Git repos, but I'm not familiar enough with their codebase to figure out what they're doing.
